src/components/ProjectIconSelection/ProjectIconSelection.js

```diff
@@ -0,0 +1,72 @@
+// @flow
+import React, { Component } from 'react';
+import styled from 'styled-components';
+import importAll from 'import-all.macro';
+
+import { shuffle } from '../../utils';
+import { BREAKPOINTS } from '../../constants';
+
+import SelectableImage from '../SelectableImage';
+
+const icons: Array<mixed> = importAll.sync(
+  '../../assets/images/icons/icon_*.*'
+);
+const iconSrcs: Array<string> = Object.values(icons).map(src => String(src));
+
+const DEFAULT_SUBSET = 21;
+const DEFAULT_ICON_SIZE = 60;
+
+type Props = {
+  selectedIcon: ?string,
+  limitTo?: number,
+  randomize?: boolean,
+  onSelectIcon: (src: string, ev: SyntheticMouseEvent<*>) => void,
+};
+
+class ProjectIconSelection extends Component<Props> {
+  randomizedIcons = shuffle(iconSrcs);
+
+  render() {
+    const { selectedIcon, limitTo, randomize } = this.props;
+
+    const iconSet = randomize ? this.randomizedIcons : iconSrcs;
+    const shownIconSubset = iconSet.slice(0, limitTo || DEFAULT_SUBSET);
```

Comment:
I have a few notes for this:

- We should use `defaultProps` for accessing this value inside the component. It's a bit cleaner, since we can omit the `|| DEFAULT_SUBSET` checks in the component, but it's also good because it ensures we're always using the same value throughout the component
- I think we should rename `DEFAULT_SUBSET` to `DEFAULT_SUBSET_LENGTH`, and we should probably just set it to the # of iconSrcs available

Here's what I imagine:

```js
const DEFAULT_SUBSET_LENGTH = iconSrcs.length;

class ProjectIconSelection extends Component<Props> {
  static defaultProps = {
    limitTo: DEFAULT_SUBSET_LENGTH,
  }

  render() {
    // omitted a few lines
    const shownIconSubset = iconSet.slice(0, limitTo);

  }
}
```